### DISCUSSION TITLE
Inverse kinematics error - first axis is not rotating

### DISCUSSION THREAD

**Discussion by MehrdadZareian** [visose/Robots#300] [Q&A] [answered]
Hi,

I tested the library with Create IRB6650 S. It goes from first point to second point with joint movement. Both points are feasible alone, but plugin can't handle them together in joint movement. Feasible points should be reachable with joint movements Am I missing something, or it is a bug?

I attached the rhino and grasshopper file for the example.

Thanks,
Mehrdad

[test.zip](https://github.com/user-attachments/files/17996754/test.zip)

**Comment by visose** [comment]
Hi, not necesarily. You are doing a cartesian motion in joint space (not an absolute joint motion). Without specifying a configuration, the plugin will choose the configuration with the shortest distance from the previous target in joint space (from the 8 posibilities). Some of these configurations might not be valid due to joint limits. What I recommend is that you specify a particular configuration to the second target. You can do this by adding the Configuration input to the Target component.

**Reply by MehrdadZareian** [reply]
Thanks a lot for the answer. It is not very clear how do you set configuration in the plugin from the documentation. 
As an example, can you tell me how do you set configuration for the attached example files?
[test2.zip](https://github.com/user-attachments/files/18005191/test2.zip)

Thank you in advance,
Mehrdad

**Reply by visose** [reply]
Select the Create Target component for the second target, right click on it, select Configuration input.

**Reply by MehrdadZareian** [reply]
Thank you for the quick response.
As you can see in the example, I have already add the configuration in the targets.  My question is how to set configuration to solve the error for this specific example.

**Reply by MehrdadZareian** [reply]
The problem I found is sometimes the shortest path between target is not working because it might choose unfeasible path. Instead, it should solve inverse kinematics for all the points and find joint angles and for simulation and collision check uses them.

**Reply by visose** [reply, marked as answer]
The plugin simulation matches the controller's path planning given the generated script file that you load to the robot. Usually the controller will give you a joint limit error in this case instead of trying to find an alternative pose/configuration that was not explicitly programmed, as this might be unexpected by the user and could cause issues / be dangerous.
